ITE/10/README.adoc

```diff
@@ -194,32 +205,30 @@ func Verify(layoutEnvelope, layoutKeys, attestations, now) {
     }
 
     for _, inspection := range layout.Inspections {
-        expectedPredicateTypes := []string{}
-        for _, expectedPredicate := range inspection.ExpectedPredicates {
-            expectedPredicateTypes = append(expectedPredicateTypes, expectedPredicate.PredicateType)
-        }
-        inspectionAttestations := executeInspection(inspection.Command, expectedPredicateTypes) // {predicateType: attestation}
-        allAttestations := combineAttestations(inspectionAttestations, attestations)
-        for _, expectedPredicate := range inspection.ExpectedPredicates {
-            attestation := inspectionAttestations[expectedPredicate.PredicateType]
-            applyMaterialRules(inspection.ExpectedMaterials, attestation.Materials, allAttestations)
-            applyProductRules(expectedPredicate.ExpectedProducts, attestation.Products, allAttestations)
-        }
+        inspectionAttestation := executeInspection(inspection.Command)
+        applyMaterialRules(inspection.ExpectedMaterials, inspectionAttestation.Materials, attestations)
+        applyProductRules(inspection.ExpectedProducts, inspectionAttestation.Products, attestations)
     }
 }
 ```
 
-[[motivation]]
-== Motivation
+=== Custom Verifiers
+
+As noted above, in-toto v1.0's verification workflow generates an in-toto
+summary link if verification succeeds. This is a powerful semantic that allows
+features like sublayouts that allow for delegating the supply chain definition
+for a step to authorized functionaries, as well as summarized verification for
+environments with limited resources. With the introduction of in-toto
+Attestation Framework, the verification workflow can be parametrized to support
+generating specific predicates as needed. For example, the default verifier can
+generate an in-toto link, another can generate SLSA Provenance, while for the
```

Comment:
Nit: I understand that verifiers might generate different predicates based on what they are checking, but I'm not sure I follow why SLSA Provenance, in particular, could be the output of a verifier.

Reply:
In some cases, SLSA provenance can be thought of as a compilation of information across multiple steps involved in building some software. For example, there may be links prior to the actual build that record environment setup or dependency provisioning. The provenance for such a build needs to include the information from such steps as well so I think we ought to explore how in-toto verification itself can collate this information across steps. There's definitely more to do here.

Reply:
Ok, thanks for clarifying. I can definitely see this sort of use case for SLSA, but the Provenance v1 seems a little less amenable to expressing non-build operations at first glance, so would definitely be interested in seeing some more concrete examples/suggestions for this.